### ISSUE TITLE
[Feature Request] Add Variant Condition for Tanks

### ISSUE THREAD

**Issue by mouse0270** [Kotori316/FluidTank#79]
I was curious if it was possible to add some sort of tag to the tanks for if they were single tank, the bottom piece, the top piece or the middle pieces, so that I could have more control over the tanks models.

Something like 
- `position: single` if a single tank.
- `position: bottom` If the tank has a tank above it
- `position: top` if the tank has a tank below it but not above it
- `position: middle` if the tank has a tank below and above it.

For example, these are my new models. The reason I am requesting this feature is because I could then use that feature in the `blockstates` to have a `tank.json`, `tank-bottom.json`, 'tank-middle.json` and `tank-top.json` so that for the middle sections you would have the break and instead the glass would go straight up to the top.

Basically, this is a feature request for a resource pack, ignore it if you like... I can live with my current changes if I have to but thought I would request it if possible.

**Comment by mouse0270** [comment]
I am going to leave this open incase you want to review it and decide if you want to include this. I actually found a mod that is basically doing this and looks almost identical to the fluid container I created in block bench.

If you don't want to include this, just close this.

**Comment by mouse0270** [comment]
So I am to stupid to figure out how to build your pull request, but if you like I can send you my models and textures that I was designing?

The new models use mostly base Minecraft textures and the glass particle. So they should scale with resource packs a lot nicer.

Here is an example, I changed star to use the obsidian portal texture/animation and then just threw in a random light-purple > light-blue > light purple animation for the creative tank.
![image](https://user-images.githubusercontent.com/564874/122770728-baae7700-d273-11eb-8c7c-433e13328725.png)

I am working on adding a little line indicators like you have on your textures but haven't gotten to that point. I am actually waiting to use your mod now, cause the one I am using looks nicer, functions worse.

**Comment by Kotori316** [comment]
Released [v16.1.7](https://www.curseforge.com/minecraft/mc-mods/largefluidtank/files/3361133)

**Comment by Kotori316** [comment]
See [here](https://github.com/Kotori316/FluidTank/blob/6dd8b717dd63d907ba2e5639b55a6a2578ab8acf/src/main/resources/assets/fluidtank/blockstates/tank_stone.json) for the available option of tanks.
I think you can use your own model by change the entry `model`.
Be careful that the renderer of content(fluid) is hardcoded so the rendering box of fluid can't be modified from resource packs.

**Comment by mouse0270** [comment]
Yeah that causes a slight visual issue as shown in this image, but I like the design of the tanks so much more.

![image](https://user-images.githubusercontent.com/564874/122967000-ff134300-d357-11eb-97da-1cb6896648ec.png)

The Gold/Diamond/Emerald tank has 251000mb of liquid in it, but since my model has a base, its not visible yet.... Not the worst thing and something I am will it give up from the nicer models.

Thanks for adding this feature so I could do this, your fluid tanks are the most flexible and I am happy to use them.

**Comment by Kotori316** [comment]
The "height" of fluid is determined by the amount of fluid in tank. It is 0.003 < height < 0.999 now. (I got some requests to show fluid even if tank has less than 0.1 % of fluid so I applied lower limit of 0.003) 
It seems that the base in your texture has 1 / 16 height, then the lower limit should be (1 / 16 + epsilon), where epsilon is a small number to avoid flicker of texture.
I'll make it possible to change the lower limit from config.
